src/main/java/com/abcdedu_backend/post/service/PostService.java

```diff
@@ -59,23 +74,30 @@ public Post checkPost(Long postId) {
         return postReposiroty.findById(postId)
                 .orElseThrow(() -> new ApplicationException(ErrorCode.POST_NOT_FOUND));
     }
+
     // post 게시자 본인과 관리자만 할 수 있는 기능에 추가
     private void checkPermission(Member member, Post post) {
         if (!member.getRole().equals(MemberRole.ADMIN) && !member.getId().equals(post.getMember().getId())) {
             throw new ApplicationException(ErrorCode.POST_INVALID_PERMISSION);
         }
     }
+    // role이 학생 이상인지
+    private void checkMemberGradeHigherThanBasic(Member member) {
+        if (member.getRole().equals(MemberRole.BASIC)) {
+            throw new ApplicationException(ErrorCode.ROLE_INVALID_PERMISSION);
+        }
+    }
```

Comment:
p4: 이부분 저도 수정한 부분이긴한데, member에서 isStudent만들어서 사용하는게 어떨까요?
```suggestion
    // role이 학생 이상인지
    private void checkMemberGradeHigherThanBasic(Member member) {
        if (!member.isStudent()) {
            throw new ApplicationException(ErrorCode.ROLE_INVALID_PERMISSION);
        }
    }
```